pandas/_libs/writers.pyx

```diff
@@ -127,19 +127,26 @@ def max_len_string_array(pandas_string[:] arr) -> Py_ssize_t:
 
     for i in range(length):
         val = arr[i]
-        if isinstance(val, str):
-            l = PyString_GET_SIZE(val)
-        elif isinstance(val, bytes):
-            l = PyBytes_GET_SIZE(val)
-        elif isinstance(val, unicode):
-            l = PyUnicode_GET_SIZE(val)
+        l = max_len_string(val)
 
         if l > m:
             m = l
 
     return m
 
 
+cpdef inline Py_ssize_t max_len_string(object val):
+    """ return the maximum size of a string"""
+    cdef:
+        Py_ssize_t l = 0
+
+    if isinstance(val, str):
```

Comment:
Do we even need this any more? Not an expert with the C API but from what I'm reading PyBytes_* is just an alias for PyString_* anyway:

https://docs.python.org/2/c-api/string.html

Reply:
Good point. Spent some 10 minutes looking at the Python 3 docs, but not very experienced with the C API, so can't really confirm if that's necessary.

The code in this `if/else` statement is present in [master in another function](https://github.com/pandas-dev/pandas/blob/e02ec8f89a743cc56a79e660b7be850448d15f61/pandas/_libs/writers.pyx#L128-L136). In this pull request I moved it to this other function to re-use it.

Reply:
I think this should actually be PyUnicode_GET_LENGTH ; PyString is part of the old Python2 interface and there may be some compat provided by Cython best I think safer to use latest and greatest API:

https://docs.python.org/3/howto/cporting.html#str-unicode-unification

https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_GET_LENGTH

Reply:
@WillAyd build is passing now :tada:  first I fixed the code without moving to `PyUnicode_GET_LENGTH`, and CI approved it. Then I changed in [this last commit](https://github.com/pandas-dev/pandas/pull/25103/commits/75246bebe144e8ec5cae3a3593062dc727728bb2) to use `PyUnicode_GET_LENGTH`. I never used it before, so hope it was the right way to add it.

- moved what's new to the new file, removing changes from the previous one (we had two releases since this PR was submitted, great work!)
- fixed formatting issues
- used `PyUnicode_GET_LENGTH`

Should be ready for review again.